### ISSUE TITLE
Missing self_loop for GCNConv when nomalize is false

### ISSUE THREAD

**Issue by elharchaoui** [pyg-team/pytorch_geometric#6251]
### 🐛 Describe the bug

Helllo,

The vanilla GCNConv does not consider adding self-loop when normalization is False, it can be visible in the implementation code of GCNConv, meanwhile here is an example that illustrates the issue.

I took the same example PyTorch geo introduction 
 [https://pytorch-geometric.readthedocs.io/en/latest/_images/graph.svg](url)

1- case without self-loop False on parameters :
```
import torch
from torch_geometric.data import Data

edge_index = torch.tensor([[0, 1, 1, 2],
                           [1, 0, 2, 1]], dtype=torch.long)

x = torch.tensor([[0], [2], [1]],
                 dtype=torch.float)

data = Data(x=x, edge_index=edge_index)


conv = GCNConv(1, 1, add_self_loops=False,
                      bias=False,
                      normalize=False,
                    aggr='max')

# force weight update for the sake of illustration

conv.lin.bias=torch.nn.Parameter(torch.tensor(0., requires_grad=True))
conv.lin.weight=torch.nn.Parameter(torch.tensor([[1.]], requires_grad=True))

with torch.no_grad():
    x1 = conv(x,edge_index)

```
result :
```
tensor([[2.],
        [1.],
        [2.]])
```
Node 1 didn't get max value 2 because there is no self loop, it's okey

2- case with self-loop True :
```
import torch
from torch_geometric.data import Data

edge_index = torch.tensor([[0, 1, 1, 2],
                           [1, 0, 2, 1]], dtype=torch.long)

x = torch.tensor([[0], [2], [1]],
                 dtype=torch.float)

data = Data(x=x, edge_index=edge_index)


conv = GCNConv(1, 1, add_self_loops=True,
                      bias=False,
                      normalize=False,
                    aggr='max')

# force weight update for the sake of illustration

conv.lin.bias=torch.nn.Parameter(torch.tensor(0., requires_grad=True))
conv.lin.weight=torch.nn.Parameter(torch.tensor([[1.]], requires_grad=True))

with torch.no_grad():
    x1 = conv(x,edge_index)
```
result :
```
tensor([[2.],
        [1.],
        [2.]])
```

Same result as before. I encountered this problem in a more complex problème, my example is simple for the illustration.

If there is something wrong in my understanding of the way of usage please let me know. thank you 

### Environment

* PyG version: 2.2.0
* PyTorch version: 1.13.0
* OS: ubuntu 
* Python version: 3.9
* CUDA/cuDNN version:  11.7
* How you installed PyTorch and PyG (`conda`, `pip`, source): conda
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by rusty1s** [comment]
Yeah, that is correct. `normalize` refers in general to the modification of the adjacency matrix inside `GCNConv`:
> normalize (bool, optional) – Whether to add self-loops and compute symmetric normalization coefficients on the fly. (default: True)

Do you think we should fix this?

**Comment by elharchaoui** [comment]
Hello, 

Thanks for your quick reaction, yes please, sometimes we need to see only the behavior of the aggregation and the linear transformation with self-loop. so yeah it will be useful to fix that.

Regards

**Comment by rusty1s** [comment]
Do you want to send a PR for this?

**Comment by elharchaoui** [comment]
Hello,

No need, i can handle this locally, thank you

Regards
